src/Marten.AsyncDaemon.Testing/cross_stream_aggregation.cs

```diff
@@ -0,0 +1,152 @@
+            events.Select(x => x.Data).OfType<IDayEvent>().All(x => x.Day == slice.Id)
+                .ShouldBeTrue();
+
+            var travels = events.OfType<Event<Travel>>().ToArray();
+            foreach (var travel in travels)
+            {
+                var index = events.As<List<IEvent>>().IndexOf(travel);
+
+                for (var i = 0; i < travel.Data.Movements.Count; i++)
+                {
+                    events.ElementAt(index + i + 1).Data.ShouldBe(travel.Data.Movements[i]);
+                }
+            }
+        }
+
+        slices.ShouldNotBeNull();
+    }
+
+    [Fact]
+    public async Task run_end_to_end()
+    {
+        StoreOptions(x => x.Projections.Add(new CrossStreamDayProjection()));
+
+        await theStore.EnsureStorageExistsAsync(typeof(Day));
+
+        using var agent = await StartDaemon();
+
+        NumberOfStreams = 10;
+        await PublishSingleThreaded();
+
+        _output.WriteLine($"Expecting {NumberOfEvents} events");
+
+        await agent.Tracker.WaitForShardState("Day:All", NumberOfEvents, 2.Minutes());
+
+        var days = await theSession.Query<Day>().ToListAsync();
+
+        var allEvents = await theSession.Events.QueryAllRawEvents().ToListAsync();
+        var dayEvents = allEvents.Select(x => x.Data).OfType<IDayEvent>();
+        var groups = dayEvents.GroupBy(x => x.Day).ToList();
+
+        foreach (var day in days)
+        {
+            var matching = groups.FirstOrDefault(x => x.Key == day.Id);
+            matching.ShouldNotBeNull();
+
+            day.Started.ShouldBe(matching.OfType<TripStarted>().Count());
+            day.Ended.ShouldBe(matching.OfType<TripEnded>().Count());
+            day.East.ShouldBe(matching
+                .OfType<Travel>()
+                .SelectMany(x => x.Movements)
+                .Where(x => x.Direction == Direction.East)
+                .Sum(x => x.Distance));
+        }
+    }
+}
+
+public class CrossStreamDayProjection: CrossStreamAggregateProjection<Day, int>
+{
+    public CrossStreamDayProjection()
+    {
+        ProjectionName = "Day";
+
+        // You have to specify this in order for the Travel events
+        // to flow through this projection.
+        IncludeType<Travel>();
+    }
+
+    protected override ValueTask GroupEvents(IEventGrouping<int> grouping, IQuerySession session, List<IEvent> events)
+    {
+        // Tell the projection how to group the events
+        // by Day document
+        grouping.AddEventsWithMetadata<IDayEvent>(e => e.Data.Day, events);
+
+        // This just lets the projection work independently
+        // on each Movement child of the Travel event
+        // as if it were its own event
+        grouping.FanOutOnEach<Travel, Movement>(x => x.Movements);
+
+        return ValueTask.CompletedTask;
+    }
+
+    public void Apply(Day day, TripStarted e) => day.Started++;
+
+    public void Apply(Day day, TripEnded e) => day.Ended++;
+
+    public void Apply(Day day, Movement e)
```

Comment:
**COMMENT:** I think that this should also be named an event, e.g. `MovementRecorded`. Showing non-event in FanOut can confuse people why we're not using event but some dto.